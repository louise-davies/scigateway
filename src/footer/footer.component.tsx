import React from 'react';
import {
  createStyles,
  StyleRules,
  Theme,
  withStyles,
  WithStyles,
} from '@material-ui/core/styles';
import { getAppStrings, getString } from '../state/strings';
import { connect } from 'react-redux';
import { StateType } from '../state/state.types';
import { AppStrings } from '../state/scigateway.types';
import { UKRITheme } from '../theming';

const styles = (theme: Theme): StyleRules =>
  createStyles({
    root: {
      position: 'absolute',
      bottom: 0,
      paddingBottom: '5px',
      width: '100%',
      fontSize: 14,
      textAlign: 'center',
      backgroundColor: theme.palette.background.default,
      '& a': {
        '&:link': {
          color: (theme as UKRITheme).link.default,
        },
        '&:visited': {
          color: (theme as UKRITheme).link.visited,
        },
        '&:active': {
          color: (theme as UKRITheme).link.active,
        },
      },
    },
  });

interface FooterProps {
  res: AppStrings | undefined;
}

export type CombinedFooterProps = FooterProps & WithStyles<typeof styles>;

const Footer = (props: CombinedFooterProps): React.ReactElement => {
  return (
    <div
      className={props.classes.root}
      dangerouslySetInnerHTML={{
        __html: getString(props.res, 'html'),
      }}
    />
  );
};

const mapStateToProps = (state: StateType): FooterProps => ({
  res: getAppStrings(state, 'footer'),
});

export const FooterWithoutStyles = Footer;
export const FooterWithStyles = withStyles(styles)(Footer);

export default connect(mapStateToProps)(FooterWithStyles);
