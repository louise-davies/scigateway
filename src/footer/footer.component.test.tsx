import React from 'react';
import { UnconnectedFooter, FooterProps } from './footer.component';
import { shallow } from 'enzyme';

describe('Footer component', () => {
  let props: FooterProps;

  beforeEach(() => {
    props = {
      res: undefined,
    };
  });

  it('footer renders correctly', () => {
    const wrapper = shallow(<UnconnectedFooter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
