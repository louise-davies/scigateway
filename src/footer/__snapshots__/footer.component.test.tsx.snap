// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component footer renders correctly 1`] = `
<Styled(div)>
  <div
    style={
      Object {
        "display": "inline-block",
        "fontSize": 14,
        "fontWeight": "bold",
        "textAlign": "left",
        "textIndent": "16px",
      }
    }
  >
    <Trans
      i18nKey="footer.links.text"
    >
      <Styled(ForwardRef(Link))
        href="footer.links.facility"
        underline="hover"
      >
        Facility Home
      </Styled(ForwardRef(Link))>
       | 
      <Styled(ForwardRef(Link))
        href="footer.links.privacy-statement"
        underline="hover"
      >
        Privacy statement
      </Styled(ForwardRef(Link))>
       | 
      <Styled(ForwardRef(Link))
        href="footer.links.data-policy"
        underline="hover"
      >
        Data policy
      </Styled(ForwardRef(Link))>
       | 
      <Styled(ForwardRef(Link))
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/accessibility"
        underline="hover"
      >
        Accessibilty statement
      </Styled(ForwardRef(Link))>
       | 
      <Styled(ForwardRef(Link))
        href="footer.links.contact"
        underline="hover"
      >
        Contact
      </Styled(ForwardRef(Link))>
    </Trans>
  </div>
  <div
    style={
      Object {
        "fontSize": "14px",
        "marginLeft": "auto",
        "paddingRight": "16px",
        "right": 0,
        "textAlign": "right",
      }
    }
  >
    <Trans
      i18nKey="footer.website-development-provider"
    >
      Built by the
      <Styled(ForwardRef(Link))
        href="https://www.scd.stfc.ac.uk/Pages/home.aspx"
        underline="hover"
      >
        Scientific Computing
      </Styled(ForwardRef(Link))>
      <Styled(ForwardRef(Link))
        href="https://www.scd.stfc.ac.uk/Pages/Software-Engineering-Group.aspx"
        underline="hover"
      >
        Data and Software Engineering Group
      </Styled(ForwardRef(Link))>
    </Trans>
  </div>
</Styled(div)>
`;
